### DISCUSSION TITLE
Accessibility

### DISCUSSION THREAD

**Discussion by ajaffary** [KaTeX/KaTeX#3120] [Ideas]
Echoing this question: https://github.com/KaTeX/KaTeX/discussions/2576#discussion-24067

Are there plans for accessibility support 'out of the box'?  I just tested the sample on the KaTeX web site with NVDA.  It doesn't see the rendered math, not even character-by-character.

I'd be happy to help build a robust accessible component.

**Comment by edemaine** [comment]
We'd be happy to review PRs that improve accessibility, but I don't think anyone is working on it at the moment.  Rereading, I think there are two approaches mentioned in #2576:
1. Somehow get the MathML visible to NVDA and other screen readers, while keeping it invisible visually.  I'm guessing our hacks used to work but don't anymore.
2. Use [render-a11y-string](https://github.com/KaTeX/KaTeX/tree/master/contrib/render-a11y-string) and apply it with an `aria-describedby` attribute, or something else that gets it read by screen readers.

The latter approach makes me wonder why we even render MathML+HTML output, if render-a11y-string can do better for accessibility and HTML is the preferred option for visual output.  If successful, it seems like we should consider bringing render-a11y-string into the default behavior, and then it would presumably replace MathML in the default behavior (though of course MathML output would remain an option).

**Comment by ronkok** [comment]
No objections here. Whatever provides the best accessibility is okay by me.
